### PR TITLE
Slice Keyboard

### PR DESCRIPTION
**Pull request by 2Moons-JP**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by 2Moons-JP** [comment]
Keyboard files are in basekeys/slice 
Only these files.

**Comment by fauxpark** [comment]
A lot of it looks copied from the Corne...

**Comment by 2Moons-JP** [comment]
03/16/2020:
matrix.c removed. Only split common will be default. 
info.json files were added for each layout of each rev.

**Comment by 2Moons-JP** [comment]
> A lot of it looks copied from the Corne...

- Physically the oled is identical to the Helix and Corne, SSD1306. The oled firmware was programmed similarly to facilitate user customization. 
- Slice PCB capability is the same as the Corne. For example RGB matrix, i2c, etc. Currently advanced features are not necessary.

File: keyboards/basekeys/slice/rev0_rgb/keymaps/2moons/keymap.c [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+oled_rotation_t oled_init_user(oled_rotation_t rotation) {
+    if (!is_keyboard_master()) {
+        return OLED_ROTATION_180;  // flips the display 180 degrees if offhand
+    }
+
+    return rotation;
+}
+
+void glcdfont(void) {
+  static const char PROGMEM crkbd_logo[] = {
```
Comment (drashna) [inline keyboards/basekeys/slice/rev0_rgb/keymaps/2moons/keymap.c]:
```suggestion
  static const char PROGMEM board_logo[] = {
```

File: keyboards/basekeys/slice/rev0_rgb/keymaps/2moons/keymap.c [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+      0x80, 0x81, 0x82, 0x83, 0x84, 0x85, 0x86, 0x87, 0x88, 0x89, 0x8a, 0x8b, 0x8c, 0x8d, 0x8e, 0x8f, 0x90, 0x91, 0x92, 0x93, 0x94,
+      0xa0, 0xa1, 0xa2, 0xa3, 0xa4, 0xa5, 0xa6, 0xa7, 0xa8, 0xa9, 0xaa, 0xab, 0xac, 0xad, 0xae, 0xaf, 0xb0, 0xb1, 0xb2, 0xb3, 0xb4,
+      0xc0, 0xc1, 0xc2, 0xc3, 0xc4, 0xc5, 0xc6, 0xc7, 0xc8, 0xc9, 0xca, 0xcb, 0xcc, 0xcd, 0xce, 0xcf, 0xd0, 0xd1, 0xd2, 0xd3, 0xd4,
+      0};
+  oled_write_P(crkbd_logo, false);
```
Comment (drashna) [inline keyboards/basekeys/slice/rev0_rgb/keymaps/2moons/keymap.c]:
```suggestion
  oled_write_P(board_logo, false);
```

File: keyboards/basekeys/slice/rev0_rgb/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+oled_rotation_t oled_init_user(oled_rotation_t rotation) {
+    if (!is_keyboard_master()) {
+        return OLED_ROTATION_180;  // flips the display 180 degrees if offhand
+    }
+
+    return rotation;
+}
+
+void glcdfont(void) {
+  static const char PROGMEM crkbd_logo[] = {
```
Comment (drashna) [inline keyboards/basekeys/slice/rev0_rgb/keymaps/default/keymap.c]:
```suggestion
  static const char PROGMEM board_logo[] = {
```

File: keyboards/basekeys/slice/rev0_rgb/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+      0x80, 0x81, 0x82, 0x83, 0x84, 0x85, 0x86, 0x87, 0x88, 0x89, 0x8a, 0x8b, 0x8c, 0x8d, 0x8e, 0x8f, 0x90, 0x91, 0x92, 0x93, 0x94,
+      0xa0, 0xa1, 0xa2, 0xa3, 0xa4, 0xa5, 0xa6, 0xa7, 0xa8, 0xa9, 0xaa, 0xab, 0xac, 0xad, 0xae, 0xaf, 0xb0, 0xb1, 0xb2, 0xb3, 0xb4,
+      0xc0, 0xc1, 0xc2, 0xc3, 0xc4, 0xc5, 0xc6, 0xc7, 0xc8, 0xc9, 0xca, 0xcb, 0xcc, 0xcd, 0xce, 0xcf, 0xd0, 0xd1, 0xd2, 0xd3, 0xd4,
+      0};
+  oled_write_P(crkbd_logo, false);
```
Comment (drashna) [inline keyboards/basekeys/slice/rev0_rgb/keymaps/default/keymap.c]:
```suggestion
  oled_write_P(board_logo, false);
```

File: keyboards/basekeys/slice/rev0_rgb/lib/host_led_state_reader.c [hunk 5]
```diff
@@ -0,0 +1,15 @@
+#include <stdio.h>
+#include "crkbd.h"
```
Comment (drashna) [inline keyboards/basekeys/slice/rev0_rgb/lib/host_led_state_reader.c]: I'm pretty sure this isn't a corne.....  :D
Also, I don't see these used in any keymap.  

All of these should be removed (except maybe the glcdfont.c file.

File: keyboards/basekeys/slice/rev1_rgb/lib/host_led_state_reader.c [hunk 6]
```diff
@@ -0,0 +1,15 @@
+#include <stdio.h>
```
Comment (drashna) [inline keyboards/basekeys/slice/rev1_rgb/lib/host_led_state_reader.c]: All of the files in the lib folder, but the glcdfont.c file can be removed.

**Comment by 2Moons-JP** [comment]
Thank you for helping me, I appreciate it. I may have made a mistake requesting a review, I'm really new at GitHub. Let me know if it's there is a better way of doing something. Thanks for everything.

**Comment by 2Moons-JP** [comment]
3/17/2020

- Rev0 and Rev0_rgb removed
- Combining of keymaps successful for all #define LAYOUT for each of the 2 keyboards.
Make all failed. Unknown how to limit which keymaps are not defined in make all

**Pull request closed without merging**